internal/http/response-recorder.go

```diff
@@ -58,6 +59,16 @@ func NewResponseRecorder(w http.ResponseWriter) *ResponseRecorder {
 	}
 }
 
+// ReadFrom implements support for calling internal io.ReaderFrom implementations
+// returns an error if the underlying ResponseWriter does not implement io.ReaderFrom
+func (lrw *ResponseRecorder) ReadFrom(r io.Reader) (int64, error) {
+	rf, ok := lrw.ResponseWriter.(io.ReaderFrom)
+	if ok {
+		return rf.ReadFrom(r)
+	}
+	return 0, errors.New("not implemented")
```

Comment:
`http.ResponseWriter` is an interface. You can assert this on creation, by doing your own interface:

```Go
// ResponseWriter extends http.ResponseWriter to always include io.ReaderFrom and flushing.
type ResponseWriter interface {
	http.ResponseWriter
	io.ReaderFrom
	http.Flusher
}

// NewResponseRecorder - returns a wrapped response writer to trap
// http status codes for auditing purposes.
func NewResponseRecorder(w ResponseWriter) *ResponseRecorder {
```

If you don't want to always have the flushing, remove it and do a proper interface check.

Edit: It seems like you are using this as a magic value. If it is not possible to assert for all wrappers (not sure why), then at least make the error an exported `ErrReadFromUnavailable` or similar.